Marsiple/App/Todos/TodosViewController.swift

```diff
@@ -101,52 +105,40 @@ private extension TodosViewController {
-            if indexPath.section == 0 {
-                reloadTableView = { [weak self] _ in
-                    self?.todosView.tableView.reloadSections([finishIndexPath.section], with: .automatic)
-                    self?.todosView.tableView.reloadRows(at: rowsToReload, with: .automatic)
-                }
-            } else {
-                reloadTableView = { [weak self] _ in
-                    self?.todosView.tableView.reloadRows(at: rowsToReload, with: .automatic)
-                    self?.todosView.tableView.reloadRows(at: [finishIndexPath], with: .automatic)
-                }
-            }
-            
-            let removedTodo = strongSelf.todos[indexPath.section].remove(at: indexPath.row)
-            removedTodo.completed = !removedTodo.completed
-            strongSelf.todos[finishIndexPath.section].insert(removedTodo, at: finishIndexPath.row)
-            strongSelf.todosView.tableView.performBatchUpdates({
-                strongSelf.todosView.tableView.moveRow(at: indexPath, to: finishIndexPath)
-            }, completion: reloadTableView)
-        }
-        return didSelectCheckBox
+    func didSelectCheckBox(atIndexPath indexPath: IndexPath) {
+        // TODO: Save changes on the server
+        let finishIndexPath = IndexPath(row: indexPath.section * todos[1 - indexPath.section].count, section: 1 - indexPath.section)
+        moveTodo(at: indexPath, to: finishIndexPath)
+        moveRow(at: indexPath, to: finishIndexPath)
     }
     
-    func indexPathsAfter(indexPath: IndexPath) -> [IndexPath] {
-        var indexPaths = [IndexPath]()
-        let maxIndexOfTodos = todos[indexPath.section].count - 2
-        guard indexPath.row <= maxIndexOfTodos else {
-            return []
-        }
-        for i in indexPath.row...maxIndexOfTodos {
-            indexPaths.append(IndexPath(row: i, section: indexPath.section))
-        }
-        return indexPaths
+    func moveTodo(at startIndexPath: IndexPath, to finishIndexPath: IndexPath) {
+        var removedTodo = todos[startIndexPath.section].remove(at: startIndexPath.row)
+        removedTodo.completed = !removedTodo.completed
+        todos[finishIndexPath.section].insert(removedTodo, at: finishIndexPath.row)
     }
     
-    func didChangeText(atIndexPath indexPath: IndexPath) -> (() -> Void)? {
-        return { [weak self] in
-            self?.todosView.tableView.reloadRows(at: [indexPath], with: .automatic)
+    func moveRow(at startIndexPath: IndexPath, to finishIndexPath: IndexPath) {
+        todosView.tableView.performBatchUpdates({
+            todosView.tableView.moveRow(at: startIndexPath, to: finishIndexPath)
+        }, completion: { [weak self] _ in
+            self?.todosView.tableView.reloadData()
+        })
+    }
+
+    func todoIsEdited(atIndexPath indexPath: IndexPath, newValue todo: Todo) {
+        // TODO: Save changes on the server
+        todos[indexPath.section][indexPath.row].title = todo.title
```

Comment:
Than the following condition would never be satisfied.

Reply:
It would be easy to do `let shouldMove = todos[indexPath.section][indexPath.row].completed != todo.completed`, that's not the problem. 

Lets leave it as-is currently.